### PR TITLE
[FEAT][alignment]: add policy and reference model wrappers

### LINKED ISSUE #14: [FEAT][alignment]: implement standardized policy and reference model wrappers
Formulate standardized Pythonic wrappers under the alignment layer to manage the active policy model and the frozen reference model instances during reinforcement learning iterations.

### PR DESCRIPTION
**Pull request by Bsgg1**
Closes #14

## Summary
- Add `rl_engine.alignment` with policy/reference model wrappers.
- Add `extract_logits(...)` for common model output shapes.
- Freeze/eval reference models by default.
- Add CPU tests for logits extraction, selected logprobs, masking, and gradients.

## Tests
- `.venv/bin/python -m pytest tests/test_alignment_model_wrappers.py -q`
- `.venv/bin/python -m pytest tests/test_reference_ops.py -q`

## Summary by CodeRabbit

* **New Features**
  * Alignment utilities expose core model-wrapping tools for RL workflows.
  * Model outputs are normalized from multiple common formats into logits.
  * Policy wrapper supports logits slicing, training-mode behavior, log-probability computation, and gradient flow.
  * Reference wrapper offers optional parameter freezing and gradient-free evaluation for reference/KL use.

* **Bug Fixes**
  * Selected-logprob computation safely handles ignored token IDs when masked out.

* **Tests**
  * Added comprehensive tests for logits extraction, wrapper behaviors, masking, and gradient semantics.

### CODE REVIEW

**Comment by Flink-ddd** [comment]
cc @inaniloquentee @EthanZero2Hero @bitborne PTAL

**Review by Flink-ddd — APPROVED** [review]
The implementation is production-ready. I’ve reviewed the logic, and the handling of gradient flows and model outputs is highly robust.

One minor architectural suggestion for the future as we scale: as we integrate with distributed backends (like FSDP), we might need to be more cautious about when we trigger freeze(). But for the current scope, this is excellent.

LGTM! I'm approving this PR.

**Review by inaniloquentee — APPROVED** [review]

**Comment by inaniloquentee** [comment]
Excellent work! thanks for contribution! Merged.
